fox_scraper/fox_scraper.py

```diff
@@ -46,41 +46,41 @@ def runner(self):
         this function will activate the scraping of the web address revieced on contruction of the class
         :return: a list of items (Item class) of the entire catalog
         """
-        main_categories_links = self.get_main_categories()
+        self.get_main_categories()
         catalog = []
-        for main_cat_num in range(len(main_categories_links)):
-            sub_categories = self.get_sub_categories(main_categories_links[main_cat_num], main_cat_num)
-            category_catalog = self.get_category_catalog(sub_categories, main_cat_num)
+        for main_category in self.main_categories_dict:
+            self.get_sub_categories(main_category)
+            category_catalog = self.get_category_catalog(main_category)
             catalog.append(category_catalog)
-        return catalog
+        self.send_all_items(catalog)
 
     def get_main_categories(self):
         """
         this function will scrap all main catalog categories
+        this function will scrap all main catalog categories
         :return: a list of beutifulSoup objects
         """
         main_categories_content = self.get_parsed_web_content(self.site_address)
         main_categories_links = []
         for link in main_categories_content.find_all(class_='link padding_hf_v '):
             main_categories_links.append(link.get('href'))
-            self.main_categories.append(link.get_text().strip())
-        return main_categories_links
+            self.main_categories_dict.update({str(link.get_text().strip()): link.get('href')})
```

Comment:
split this line, use variables and I still don't understand why are you converting string to string